rfd/0180-sso-as-mfa-method.md

```diff
@@ -0,0 +1,425 @@
+}
+
+message SAMLConnectorSpecV2 {
+  ...
++  // AllowAsMFAMethod represents whether this auth connector can be used as an MFA
++  // method or not.
++  AllowAsMFAMethod AllowAsMFAMethod = 17 [(gogoproto.jsontag) = "allow_as_mfa_method,omitempty"];
+}
+ 
+message GithubConnectorSpecV3 {
+  ...
++  // AllowAsMFAMethod represents whether this auth connector can be used as an MFA
++  // method or not.
++  AllowAsMFAMethod AllowAsMFAMethod = 10 [(gogoproto.jsontag) = "allow_as_mfa_method,omitempty"];
+}
+```
+
+**CreatePrivilegedToken**
+```diff
+message OIDCAuthRequest {
+  ...
++  // CreatePrivilegedToken is an option to create a privileged token instead of creating 
++  // a user session. Privileged tokens can be used in place of standard MFA verification for
++  // privileged actions. This action is only allowed if the auth connector is allowed
++  // to be used as an MFA method and if the user is pre-authenticated (not first time login).
++  bool CreatePrivilegedToken = 19 [(gogoproto.jsontag) = "create_privileged_token,omitempty"];
+}
+
+message SAMLAuthRequest {
+  ...
++  // CreatePrivilegedToken is an option to create a privileged token instead of creating 
++  // a user session. Privileged tokens can be used in place of standard MFA verification for
++  // privileged actions. This action is only allowed if the auth connector is allowed
++  // to be used as an MFA method and if the user is pre-authenticated (not first time login).
++  bool CreatePrivilegedToken = 18 [(gogoproto.jsontag) = "create_token,omitempty"];
+}
+
+message GithubAuthRequest {
+  ...
++  // CreatePrivilegedToken is an option to create a privileged token instead of creating 
++  // a user session. Privileged tokens can be used in place of standard MFA verification for
++  // privileged actions. This action is only allowed if the auth connector is allowed
++  // to be used as an MFA method and if the user is pre-authenticated (not first time login).
++  bool CreatePrivilegedToken = 18 [(gogoproto.jsontag) = "create_token,omitempty"];
+}
+```
+
+**AuthConnectorChallenge**
+```diff
+message MFAAuthenticateChallenge {
+  ...
++  // MFAConnectorChallenge is an auth connector MFA challenge. If set, the client
++  // will attempt to create an auth request with this connector to acquire a
++  // privileged token as a substitute for local MFA.
++  MFAConnectorChallenge MFAConnectorChallenge = 5;
+}
+
++// MFAConnectorChallenge contains auth connector details for for completing an
++// auth connector MFA challenge.
++message MFAConnectorChallenge {
++  // Type is the auth connector type.
++  string Type = 1;
++  // ID is the auth connector ID.
++  string ID = 2;
++}
+
+message MFAAuthenticateResponse {
+  oneof Response {
+    ...
++    // TokenID is a privileged token ID used as a substitute for local MFA.
++    string TokenID = 4;
+  }
+}
+```
+
+### Backward Compatibility
+
+N/A
+
+### Audit Events
+
+TODO
```

Comment:
At a minimum, I'd expect the `mfa_device` field in existing audit events to be traceable to the connector used for MFA.